tests/integration/build_test.sh

```diff
@@ -0,0 +1,189 @@
+  fi
+
+  optimize_dot_config "$container" "${container_tmp_dir}/linux"
+}
+
+# Optimize the .config file in a container.
+#
+# @container          The ID or name of the container.
+# @container_tmp_dir  The temporary directory in the container to use for intermediate files.
+function optimize_dot_config()
+{
+  local container="$1"
+  local container_tmp_dir="$2"
+
+  # Generate a list of currently loaded modules in the container
+  container_exec "$container" "cd ${container_tmp_dir} && /usr/sbin/lsmod > container_mod_list"
+  if [[ "$?" -ne 0 ]]; then
+    fail "(${LINENO}): Failed to generate module list in container."
+  fi
+
+  # Create a default configuration and then update it to reflect current settings
+  container_exec "$container" "cd ${container_tmp_dir} && make defconfig && make olddefconfig"
+  if [[ "$?" -ne 0 ]]; then
+    fail "(${LINENO}): Failed to create default configuration in container."
+  fi
+
+  # Optimize the configuration based on the currently loaded modules
+  container_exec "$container" "cd ${container_tmp_dir} && make LSMOD=${container_tmp_dir}/container_mod_list localmodconfig"
+  if [[ "$?" -ne 0 ]]; then
+    fail "(${LINENO}): Failed to optimize configuration based on loaded modules in container."
+  fi
+}
+
+# This function tests the kernel build process in specified container environments
+function test_kernel_build()
+{
+  local container
+  local distro
+  local container_tmp_dir
+  local kw_build_cmd
+  local build_status
+  local build_result
+  local build_status_log
+
+  # Define the container name using the target random distribution
+  container="kw-${target_random_distro}"
+  # Create a temporary directory inside the container
+  container_tmp_dir=$(container_exec "$container" 'mktemp --directory')
+
+  # Prepare the environment inside the container
+  prepare_environment "$container" "$container_tmp_dir"
+
+  kw_build_cmd='kw build'
+  container_exec "$container" "cd ${container_tmp_dir}/linux && ${kw_build_cmd}"
+  assert_equals_helper "kw build failed for ${container}" "($LINENO)" 0 "$?"
+
+  # Retrieve the build status log from the database
+  build_status_log=$(container_exec "$container" "sqlite3 ~/.local/share/kw/kw.db \"SELECT * FROM statistics_report\" | tail --lines=1")
+
+  # Extract the build status and result from the log
+  build_status=$(printf '%s' "$build_status_log" | cut -d'|' -f2)
+  assert_equals_helper "Build status check failed for $container" "$LINENO" 'build' "$build_status"
+
+  build_result=$(printf '%s' "$build_status_log" | cut -d'|' -f3)
+  assert_equals_helper "Build result check failed for $container" "$LINENO" 'success' "$build_result"
+
+  # Clean the kernel tree for the next test
+  container_exec "$container" "cd ${container_tmp_dir}/linux && kw build --clean"
+  assert_equals_helper "kw build --clean failed for ${container}" "($LINENO)" 0 "$?"
+}
+
+function test_kernel_build_only_cpu_scaling_option()
```

Comment:
I was thinking about this test, and you should find a way to identify CPU utilization when compiling the kernel. If it is around half of the CPU capacity, it should pass. You don't even need to run the entire test for that.